src/usb-device-node.js

```
@@ -89,25 +89,53 @@ class UsbDevice {
 		});
 	}
 
-	transferOut(setup, data) {
+	transferOut(setup, reqData) {
 		return new Promise((resolve, reject) => {
-			if (!data) {
+			if (!reqData) {
 				// NOTE: this is a quirk of some USB device-side implementations
 				// where zero-length (no data stage) OUT control requests are not
 				// completed and cause a timeout.
 				// The workaround is always including a data stage.
 				if (!this._quirks.controlOutTransfersRequireDataStage) {
-					data = Buffer.alloc(0);
+					reqData = Buffer.alloc(0);
 				} else {
-					data = Buffer.alloc(1);
+					reqData = Buffer.alloc(1);
 				}
 			}
-			this._dev.controlTransfer(setup.bmRequestType, setup.bRequest, setup.wValue, setup.wIndex, data, err => {
-				if (err) {
-					return reject(wrapUsbError(err, 'OUT control transfer failed'));
-				}
-				resolve();
-			});
+			this._dev.controlTransfer(
+				setup.bmRequestType,
+				setup.bRequest,
+				setup.wValue,
+				setup.wIndex,
+				reqData, (err, bytes) => {
+					if (err) {
+						return reject(wrapUsbError(err, 'OUT control transfer failed'));
+					}
+					resolve(bytes);
+				});
+		});
+	}
+
+	getDescriptorString(intrface) {
+		return new Promise((resolve, reject) => {
+			try {
+				this._dev.getStringDescriptor(intrface, (err, intrfaceName) => {
+					if (err) {
+						try {
+							this._dev.close();
+						} catch (err) {
+							this._log.error(`Unable to close device: ${err.message}`);
+							// Ignore error
+						}
+						reject(wrapUsbError(err, 'Unable to get serial number descriptor'));
+						return;
+					}
+					this._dev.particle.isOpen = true;
+					resolve(intrfaceName);
+				});
+			} catch (err) {
+				reject(wrapUsbError(err, 'Failed to claim interface'));
+			}
```

The error handling seems to be copied from the `open()` method but it doesn't make sense in this context. We should just return the error if getting the string descriptor fails.